packages/backend/src/apps/custom-api/common/add-headers.ts

```diff
@@ -3,11 +3,10 @@ import { TBeforeRequest } from '@plumber/types'
 const addHeaders: TBeforeRequest = ($, requestConfig) => {
   const authData = $.auth.data
   if (authData?.headers) {
-    requestConfig.headers = {
-      ...requestConfig.headers,
-      'Content-Type': 'application/json',
-      ...(authData.headers as Record<string, string>),
-    }
+    requestConfig.headers['Content-Type'] = 'application/json'
```

Comment:
nit: this one can also use `set`?
```suggestion
    requestConfig.headers.set('Content-Type', 'application/json')
```